test/testenv/appframework_utils.go

```diff
@@ -311,3 +329,91 @@ func WaitforPhaseChange(deployment *Deployment, testenvInstance *TestEnv, name s
+		for _, appSource := range appSource {
+			testenvInstance.Log.Info(fmt.Sprintf("Verify apps '%v' state on CR %v with name %v", phase, appSource.CrKind, appSource.CrName))
+			VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appSource.CrAppFileList)
+		}
+	}
+
+	// Verify apps packages are deleted from the operator pod for all CRs
+	opPod := GetOperatorPodName(testenvInstance.GetName())
+	for _, appSource := range appSource {
+		testenvInstance.Log.Info(fmt.Sprintf("Verify apps %s packages are deleted from the operator pod for CR %v with name %v", appSource.CrAppVersion, appSource.CrKind, appSource.CrName))
+		opPath := filepath.Join(splcommon.AppDownloadVolume, "downloadedApps", testenvInstance.GetName(), appSource.CrKind, deployment.GetName(), appSource.CrAppScope, appSource.CrAppSourceName)
+		VerifyAppsPackageDeletedOnContainer(deployment, testenvInstance, testenvInstance.GetName(), []string{opPod}, appSource.CrAppFileList, opPath)
+	}
+
+	// Verify apps 'install' state for all CRs
+	for _, appSource := range appSource {
+		testenvInstance.Log.Info(fmt.Sprintf("Verify apps '%v' state on CR %v with name %v", enterpriseApi.PhaseInstall, appSource.CrKind, appSource.CrName))
+		VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, enterpriseApi.PhaseInstall, appSource.CrAppFileList)
+	}
+
+	// Verify apps packages are deleted from the CR pods
+	for _, appSource := range appSource {
+		podDownloadPath := "/init-apps/" + appSource.CrAppSourceVolumeName
+		pod := appSource.CrPod
+		testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps packages are deleted on pod %s", appSource.CrAppVersion, pod))
+		VerifyAppsPackageDeletedOnContainer(deployment, testenvInstance, testenvInstance.GetName(), pod, appSource.CrAppFileList, podDownloadPath)
+	}
+
+	// Verify bundle push status
+	for _, appSource := range appSource {
+		if appSource.CrKind == "ClusterMaster" && appSource.CrAppScope == enterpriseApi.ScopeCluster {
+			testenvInstance.Log.Info(fmt.Sprintf("Verify Cluster Manager bundle push status (%s apps) and compare bundle hash with previous bundle hash", appSource.CrAppVersion))
+			VerifyClusterManagerBundlePush(deployment, testenvInstance, testenvInstance.GetName(), appSource.CrReplicas, clusterManagerBundleHash)
+			if clusterManagerBundleHash == "" {
+				clusterManagerBundleHash = GetClusterManagerBundleHash(deployment)
+			}
+		}
+		if appSource.CrKind == "SearchHeadCluster" && appSource.CrAppScope == enterpriseApi.ScopeCluster {
+			testenvInstance.Log.Info(fmt.Sprintf("Verify Deployer bundle push status (%s apps)", appSource.CrAppVersion))
+			VerifyDeployerBundlePush(deployment, testenvInstance, testenvInstance.GetName(), appSource.CrReplicas)
+		}
+	}
+
+	// Verify apps are copied to correct location on all CRs
+	for _, appSource := range appSource {
+		allPodNames := appSource.CrPod
```

Comment:
Nit: We don't need this additional variable `allPodNames` when we can directly use appSource.CrPod in the caller for `VerifyAppsCopied`